chall06/ohakola.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+import sys
+
+# Loops through books and tries to fit them to shelves
+# If count goes too high, return -1
+def shelves_needed(books, shelves, shelf_index, book_index):
+	count = 1
+	curr_shelf = shelves[shelf_index]
+	books_len = len(books)
+	shelves_len = len(shelves)
+	books_placed = 0
+	while books_placed < books_len:
+		if curr_shelf >= books[book_index]:
+			curr_shelf = curr_shelf - books[book_index]
+			book_index += 1
+			if book_index >= books_len:
+				book_index = 0
+			books_placed += 1
+		else:
+			count += 1
+			if count > shelves_len:
+				return -1
+			shelf_index += 1
+			if shelf_index >= shelves_len:
+				shelf_index = 0
+			curr_shelf = shelves[shelf_index]
+	return count
+
+# Sorts shelves from largest to smalles after which attempt shelves_needed
```

Comment:
As your comment describes the purpose of your function overall, you should prefer documenting it with a docstring. On the other hand, you can add comments to more specific parts of the function that could not be obvious. For example, you could briefly comment on your usage of the `MAX_INT`.